MSAL/src/MSALPublicClientApplication.m

```diff
@@ -739,15 +774,35 @@ - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
     params.loginHint = loginHint;
     params.extraAuthorizeURLQueryParameters = extraQueryParameters;
     params.accountIdentifier = account.lookupAccountIdentifier;
-    params.validateAuthority = _validateAuthority;
-    params.extraURLQueryParameters = _sliceParameters;
-    params.tokenExpirationBuffer = _expirationBuffer;
-    params.extendedLifetimeEnabled = _extendedLifetimeEnabled;
-    params.clientCapabilities = _clientCapabilities;
+    
+    params.extraURLQueryParameters = _configuration.extraQueryParameters.extraURLQueryParameters;
 
+    NSMutableDictionary *extraAuthorizeURLQueryParameters = [_configuration.extraQueryParameters.extraAuthorizeURLQueryParameters mutableCopy];
+    [extraAuthorizeURLQueryParameters addEntriesFromDictionary:extraQueryParameters];
+    params.extraAuthorizeURLQueryParameters = extraAuthorizeURLQueryParameters;
+    params.extraTokenRequestParameters = _configuration.extraQueryParameters.extraTokenURLParameters;
+    
+    params.tokenExpirationBuffer = _configuration.tokenExpirationBuffer;
+    params.extendedLifetimeEnabled = _configuration.extendedLifetimeEnabled;
+    params.clientCapabilities = _configuration.clientApplicationCapabilities;
+
+    params.validateAuthority = _configuration.authority.validateAuthority;
+    
+    if (params.validateAuthority && MSALGlobalConfig.knownAuthorities)
+    {
+        for (MSALAuthority *knownAuthority in MSALGlobalConfig.knownAuthorities)
+        {
+            if ([params.authority isKindOfClass:knownAuthority.class]
+                && [knownAuthority.url isEqual:params.authority.url])
+            {
+                params.validateAuthority = NO;
```

Comment:
For today's meeting, for AAD authorities we should always validate them.

Reply:
Unless it's a known authority, no?

Reply:
I think even if it's a known authority. Let's follow up offline.